src/cargo/core/resolver/mod.rs

```diff
@@ -1164,7 +1291,15 @@ fn activate_deps_loop(
                             past.push(conflicting_activations.clone());
                         }
                     }
-                    // if not has_another we we activate for the better error messages
+
+                    // If we don't have any deps known to fail then we enter
+                    // here and push our frame onto the list of frames to
+                    // resolve.
+                    //
+                    // If a dependency is known to fail we may want to skip this
+                    // frame as this is surely a dead end. If there are more
+                    // candidates we wait until the last one to generate an
+                    // error message in this case.
```

Comment:
This is better than what I had, by a lot, but I wonder if it could be clearer. Semse like a lot of interacting clauses in the last sentence. What is `in this case` modifying?
Maybe:

If a dependency is known to fail we want to skip this frame as this is surely a dead end. If there are more candidates then we can skip this frame without affecting error messages. However, if this is last candidate we activate to get its error message.

Reply:
Perfect!